src/renderer/components/message/MessageWrapper.tsx

```diff
@@ -1,17 +1,64 @@
-import React from 'react'
+import React, { useLayoutEffect } from 'react'
 import Message from './Message'
 import { ConversationType } from './MessageList'
 import { Type } from '../../backend-com'
+import { C } from 'deltachat-node/node/dist/constants'
+import { getLogger } from '../../../shared/logger'
+
+const log = getLogger('renderer/message/MessageWrapper')
 
 type RenderMessageProps = {
+  key2: string
   message: Type.Message
   conversationType: ConversationType
+  unreadMessageInViewIntersectionObserver: React.MutableRefObject<IntersectionObserver | null>
 }
 
 export const MessageWrapper = (props: RenderMessageProps) => {
+  const state = props.message.state
+  const shouldInViewObserve =
+    state === C.DC_STATE_IN_FRESH || state === C.DC_STATE_IN_NOTICED
```

Comment:
Also not sure about this one? If this bool is true we observe that message if it's in the viewport or not